### PR TITLE
Transform IntCon(0) to VecCon(0) for WithElement case - fixes emitter assertion for crossgen2

### LINKED ISSUE #81963: Crossgen2 asserts in JIT when compiling The JIT\Regression\JitBlue\Runtime_81739
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=167278
Build error leg or test failing: JIT\\Regression\\JitBlue\\Runtime_81739\\Runtime_81739\\Runtime_81739.cmd
Pull request: https://github.com/dotnet/runtime/pull/81956

## Error Message

" "dotnet" "C:\h\w\ACF5090A\p\crossgen2\crossgen2.dll" @"C:\h\w\ACF5090A\w\AA3A0904\e\JIT\Regression\JitBlue\Runtime_81739\Runtime_81739\Runtime_81739.dll.rsp"   -r:C:\h\w\ACF5090A\w\AA3A0904\e\JIT\Regression\JitBlue\Runtime_81739\Runtime_81739\IL-CG2\*.dll"
D:\a\_work\1\s\src\coreclr\jit\emitxarch.cpp:6034
Assertion failed 'isFloatReg(dstReg) && isFloatReg(srcReg)' in 'Runtime_81739:Main():int' during 'Generate code' (IL size 27; hash 0x0dd256ab; FullOpts)

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

### PR DESCRIPTION
**Pull request by TIHan**
Fixes https://github.com/dotnet/runtime/issues/81963
This fixes the assertion that was occurring in the emitter recently.
Thanks to @tannergooding for his help on this.

### CODE REVIEW

File: src/coreclr/jit/lclmorph.cpp [hunk 1]
```diff
@@ -1209,13 +1209,21 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
                     assert(elementType == TYP_SIMD12);
                     assert(varDsc->TypeGet() == TYP_SIMD16);
 
+                    // If we are not doing struct promotion and we have zero-initialization,
+                    // then we need to produce a VecCon(0).
+                    if (elementNode->IsIntegralConst(0))
```
Comment (tannergooding) [inline src/coreclr/jit/lclmorph.cpp:1214]: Just wanted to note that we looked at fixing `CEE_INITOBJ` to do have `TYP_SIMD` be an `ASG` of `ZeroConNode` like it is for primitives

But, it still asserted so there are likely other places also introducing the `ASG = 0` that would also need to be fixed first.

**Comment by TIHan** [comment]
@dotnet/jit-contrib this is ready and appears to fix the assertion based on the outerloop run.

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**